scripts/jenkins-common.sh

```diff
@@ -24,9 +31,29 @@ fi
 # Activate the Python virtualenv
 source $HOME/edx-venv/bin/activate
 
-# add the node_js packages dir to PATH
+# add the node packages dir to PATH
 PATH=$PATH:node_modules/.bin
 
+echo "setting up nodeenv"
+pip install nodeenv
+# Ensure we are starting with a clean node env directory
+rm -rf $NODE_ENV_DIR
+WAIT_COUNT=0
+until timeout doCreateNodeEnv || [ $WAIT_COUNT -eq 3 ]; do
```

Comment:
if `doCreateNodeEnv` itself exits with nonzero status, your loop would assume a timeout.  May need to restructure the loop to accomodate for that detail.  timeout(1) will exit with status 127 on timeout, which is what you're really looking for.